### PR TITLE
Compatibility to draft 07

### PR DESCRIPTION
**Pull request by netanel-mce**
**what is here**

- new flag `compatibility` to choose if output schema must to match to draft 07, if `draft-07` is chosen, so:
   - `undefined` is not a valid for `definition.type`, so: 
      - if main type is alias of undefined - we will throw error
      - if have a property undefined - we will ignore (like never)
   - `symbol` is not a valid for `definision.type`, so we write `object` instead
   - we not using in the operator `typeof` for `definition.type` only after check what returned from he
- new type `DefinitionTypeName` for save on `definition.type`
- unit test with compatibility `draft-07` and `none`
- updating README.md file
- updating api.md file
- fix issue: #490

### CODE REVIEW

**Comment by domoritz** [comment]
I'm not following. Are you saying that we get errors if we generate non-standard schemas and that we currently do create those schemas? In that case I would strongly vote for making standard compliant output the default without a switch.

**Comment by netanel-mce** [comment]
@domoritz Yes, that's the situation today, I'll remove the switch

**Comment by netanel-mce** [comment]
> I'm not following. Are you saying that we get errors if we generate non-standard schemas and that we currently do create those schemas? In that case I would strongly vote for making standard compliant output the default without a switch.

@domoritz done, all schemas should compatibility to draft-07.

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Options:
   --id                  Set schema id.                                               [string]  [default: ""]
   --defaultNumberType   Default number type.                                         [choices: "number", "integer"] [default: "number"]
   --tsNodeRegister      Use ts-node/register (needed for require typescript files).  [boolean] [default: false]
+  --compatibility       Output compatibility mode                                    [choices: "none", "draft-07"] [default: "none"]
```
Comment (domoritz) [inline README.md:54]: This is gone, right?

**Review by domoritz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (netanel-mce) [reply README.md]: @domoritz right!, done.

**Comment by domoritz** [comment]
Thank you

**Pull request merged**